packages/workflow/src/features/correction/routes.ts

```diff
@@ -293,28 +298,56 @@ export const routes = [
       const { correction, ...registration } =
         recordInputWithUploadedAttachments.registration!
 
+      /*
+       * Separate tasks from the record
+       */
+      const [correctionRequestApprovedTask, correctedTask] =
+        getTasksInAscendingOrder(recordInCorrectedState)
+      const recordWithoutTasks = getRecordWithoutTasks(recordInCorrectedState)
+
+      /*
+       * Create and update the valid record that only has one task (the corrected task)
+       */
+      const newRecord = addTaskToRecord(recordWithoutTasks, correctedTask)
+
+      newRecord satisfies ValidRecord
+
       const recordWithUpdatedValues = updateFHIRBundle(
-        recordInCorrectedState,
```

Comment:
This was what  I think caused the issue. Bundle with 2 tasks got into the FHIR updater and the "old" approve task got its `lastModified` changed because the correction data on line after this one contains "createdAt" field, which modifies it